scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/SparkeyUriSideInput.scala

```diff
@@ -0,0 +1,66 @@
+package com.spotify.scio.extra.sparkey
+
+import java.util
+
+import com.spotify.scio.ScioContext
+import com.spotify.scio.io.{EmptyTap, EmptyTapOf, ScioIO, Tap, TapOf}
+import com.spotify.scio.util.RemoteFileUtil
+import com.spotify.scio.values.SCollection
+import com.spotify.sparkey.{IndexHeader, LogHeader, SparkeyReader}
+import org.apache.beam.sdk.options.PipelineOptions
+import com.google.protobuf.ByteString
+
+
+object SparkeyUriSideInput {
+  object ReadP {
+    def apply(options: PipelineOptions): ReadP = ReadP(() => RemoteFileUtil.create(options))
+  }
+  case class ReadP(rfuCreateFn: () => RemoteFileUtil)
+
+  private trait TestSparkeyReader extends SparkeyReader {
+    override def getAsString(key: String): String = ???
+    override def getAsByteArray(key: Array[Byte]): Array[Byte] = ???
+    override def getAsEntry(key: Array[Byte]): SparkeyReader.Entry = ???
+    override def getIndexHeader: IndexHeader = ???
+    override def getLogHeader: LogHeader = ???
+    override def duplicate(): SparkeyReader = ???
+    override def close(): Unit = ???
+    override def iterator(): util.Iterator[SparkeyReader.Entry] = ???
+  }
+
+  def forTest(items: Map[String, String]): SparkeyUri = new SparkeyUri {
+    override val basePath: String = "i should be ignored"
+    override private[sparkey] def exists = true
+    override def getReader: SparkeyReader = new TestSparkeyReader {
+      override def getAsString(key: String): String = items.get(key).orNull
+    }
+  }
+
+  def forTest(items: Seq[(Array[Byte], Array[Byte])]): SparkeyUri = new SparkeyUri {
+    private val byteStrItems = items.map { case (k, v) => ByteString.copyFrom(k) -> v }.toMap
+    override val basePath: String = "i should be ignored"
+    override private[sparkey] def exists = true
+    override def getReader: SparkeyReader = new TestSparkeyReader {
+      override def getAsByteArray(key: Array[Byte]): Array[Byte] = {
+        byteStrItems.get(ByteString.copyFrom(key)).orNull
+      }
+    }
+  }
+}
+
+final case class SparkeyUriSideInput(path: String) extends ScioIO[SparkeyUri] {
```

Comment:
I find this naming confusing 😬 also having users supplying test data through `forTest` does not conform with other IO's. This leads me to believe that this might not be 1) exactly what we want 2) some more deeper changes are needed to support these use cases. In any case imo this requires a bit more thought.